src/CommonLib/Processors/LDAPPropertyProcessor.cs

```diff
@@ -148,45 +106,84 @@ public async Task<UserProperties> ReadUserProperties(ISearchResultEntry entry)
-            if (int.TryParse(uac, out var flag))
-            {
-                var flags = (UacFlags)flag;
-                enabled = (flags & UacFlags.AccountDisable) == 0;
-                trustedToAuth = (flags & UacFlags.TrustedToAuthForDelegation) != 0;
-                sensitive = (flags & UacFlags.NotDelegated) != 0;
-                dontReqPreAuth = (flags & UacFlags.DontReqPreauth) != 0;
-                passwdNotReq = (flags & UacFlags.PasswordNotRequired) != 0;
-                unconstrained = (flags & UacFlags.TrustedForDelegation) != 0;
-                pwdNeverExpires = (flags & UacFlags.DontExpirePassword) != 0;
-            }
-            else
-            {
-                trustedToAuth = false;
-                enabled = true;
-                sensitive = false;
-                dontReqPreAuth = false;
-                passwdNotReq = false;
-                unconstrained = false;
-                pwdNeverExpires = false;
-            }
+            userProps.Props = props;
 
-            props.Add("sensitive", sensitive);
-            props.Add("dontreqpreauth", dontReqPreAuth);
-            props.Add("passwordnotreqd", passwdNotReq);
-            props.Add("unconstraineddelegation", unconstrained);
-            props.Add("pwdneverexpires", pwdNeverExpires);
-            props.Add("enabled", enabled);
-            props.Add("trustedtoauth", trustedToAuth);
-            var domain = Helpers.DistinguishedNameToDomain(entry.DistinguishedName);
+            return userProps;
+        }
+
+        /// <summary>
+        ///     Reads specific LDAP properties related to Computers
+        /// </summary>
+        /// <param name="entry"></param>
+        /// <returns></returns>
+        public async Task<ComputerProperties> ReadComputerProperties(ISearchResultEntry entry)
+        {
+            var compProps = new ComputerProperties();
+            var props = GetCommonProps(entry);
+            
+            props.AddRange(GetProperties(LDAPProperties.UserAccountControl, entry));
+
+            props.AddRange((GetProperties(LDAPProperties.AllowedToDelegateTo, entry)));
+            compProps.AllowedToDelegate = (await ReadPropertyDelegates(entry)).ToArray();
+            
+            compProps.AllowedToAct = ReadAllowedToActPrinciples(entry).ToArray();
+
+            props.AddRange(GetProperties(LDAPProperties.LastLogon, entry));
+            props.AddRange(GetProperties(LDAPProperties.LastLogonTimestamp, entry));
+            props.AddRange(GetProperties(LDAPProperties.PasswordLastSet, entry));
+            props.AddRange(GetProperties(LDAPProperties.ServicePrincipalNames, entry));
+            props.AddRange(GetProperties(LDAPProperties.Email, entry));
+            props.AddRange(GetProperties(LDAPProperties.OperatingSystem, entry));
+
+            var sidHistory = ReadSidHistory(entry);
+            props.Add(PropertyMap.GetPropertyName(LDAPProperties.SIDHistory), sidHistory.History.ToArray());
```

Comment:
SidHistory breaks the new design a little because it provides dual functionality that can't be split without duplicating efforts (see `ReadSidHistory` below), and so we map the property value here instead of within `GetProperties` as the others.  Not sure how I want to resolve this.

Reply:
Cleaned this up a bit.